[CI] RelocationIT testRelocationEstablishedPeerRecoveryRetentionLeases failure

```
java.lang.AssertionError: 
Allocated on new node: node_t3
Expected: is <true>
     but: was <false>

at __randomizedtesting.SeedInfo.seed([F2359E39E52623FA:328593C56CC98FC5]:0)
at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
at org.junit.Assert.assertThat(Assert.java:956)
at org.elasticsearch.test.ESIntegTestCase.assertAllShardsOnNodes(ESIntegTestCase.java:2045)
at org.elasticsearch.recovery.RelocationIT.lambda$testRelocationEstablishedPeerRecoveryRetentionLeases$13(RelocationIT.java:637)
at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:904)
at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:877)
at org.elasticsearch.recovery.RelocationIT.testRelocationEstablishedPeerRecoveryRetentionLeases(RelocationIT.java:637)
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk13,ES_RUNTIME_JAVA=corretto8,nodes=general-purpose/433/console
https://gradle-enterprise.elastic.co/s/bnjckkmcp4zcs

There a have been a few occurrences of this failure all the 7.x branch

https://gradle-enterprise.elastic.co/s/eycc3fznn6y2k
https://gradle-enterprise.elastic.co/s/2ahivt5b2mlwq
https://gradle-enterprise.elastic.co/s/qgilyttuh5izi

--- a/server/src/test/java/org/elasticsearch/recovery/RelocationIT.java
+++ b/server/src/test/java/org/elasticsearch/recovery/RelocationIT.java
@@ -622,7 +622,8 @@ public class RelocationIT extends ESIntegTestCase {
             Stream.generate(() -> Settings.builder().put("node.attr.color", "red").build()).limit(halfNodes)).toArray(Settings[]::new);
         List<String> nodes = internalCluster().startNodes(nodeSettings);
         String[] blueNodes = nodes.subList(0, halfNodes).toArray(String[]::new);
-        String[] redNodes = nodes.subList(0, halfNodes).toArray(String[]::new);
+        String[] redNodes = nodes.subList(halfNodes, nodes.size()).toArray(String[]::new);
+        logger.debug("--> blue nodes: [{}], red nodes: [{}]", blueNodes, redNodes);
         ensureStableCluster(halfNodes * 2);
         assertAcked(
             client().admin().indices().prepareCreate(indexName).setSettings(Settings.builder()

